### ISSUE TITLE
Master host IP is not removed from the Nodes Section 

### ISSUE THREAD

**Issue by philipakash** [aws-quickstart/quickstart-redhat-openshift#108]
When the first time after the installation master is deleted by ASG, the OSE scaler won't remove old master host(one which got deleted) from the nodes section(initial inventory adds all the masters to nodes) .

**Comment by philipakash** [comment]
Logs and Inventory :

Before deleting 10.76.10.221
```
[OSEv3:children]
masters
nodes
etcd
new_nodes
new_masters
new_etcd

[OSEv3:vars]
ansible_ssh_user=root
openshift_deployment_type=openshift-enterprise
containerized=false
openshift_disable_check=package_version
openshift_master_cluster_method=native
openshift_router_selector='region=infra'
openshift_registry_selector='region=infra'
openshift_master_default_subdomain=***********
openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]
openshift_check_min_host_memory_gb=6
openshift_master_cluster_hostname=**********
openshift_master_cluster_public_hostname=*************

[etcd]
10.76.12.45     openshift_hostname=10.76.12.45                                        
10.76.11.112    openshift_hostname=10.76.11.112                                        
10.76.10.240    openshift_hostname=10.76.10.240                                    

[masters]
10.76.11.183    openshift_hostname=10.76.11.183                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 
**10.76.10.221    openshift_hostname=10.76.10.221                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false** 
10.76.12.116    openshift_hostname=10.76.12.116                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 

[nodes]
10.76.11.183    openshift_hostname=10.76.11.183                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 
10.76.10.221    openshift_hostname=10.76.10.221                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 
10.76.10.59     openshift_hostname=10.76.10.59                                      openshift_node_labels="{'application_node': 'yes', 'registry_node': 'yes', 'router_node': 'yes', 'region': 'infra', 'zone': 'default'} "  
10.76.12.209    openshift_hostname=10.76.12.209                                      openshift_node_labels="{'application_node': 'yes', 'registry_node': 'yes', 'router_node': 'yes', 'region': 'infra', 'zone': 'default'} "  
10.76.11.176    openshift_hostname=10.76.11.176                                      openshift_node_labels="{'application_node': 'yes', 'registry_node': 'yes', 'router_node': 'yes', 'region': 'infra', 'zone': 'default'} "  

[new_nodes]

[provision_in_progress]

[new_masters]

[new_etcd]

```
After deleting 10.76.10.221
After Termination Logs:
```
2018-04-07T11:22:28UTC scaler.py  INFO    : Verifying that the API reflects the scaling events properly
2018-04-07T11:22:28UTC scaler.py  INFO    : Great! The API contains scaling events that we need to process!
2018-04-07T11:22:28UTC utils.py   INFO    : [wait_for_api]: Waiting for the EC2 API to return new instances.
2018-04-07T11:22:28UTC utils.py   INFO    : [wait_for_api] Complete
2018-04-07T11:22:28UTC utils.py   INFO    : We're processing the scaling pipeline
2018-04-07T11:22:28UTC utils.py   INFO    : We have the following nodes to remove from the inventory:
2018-04-07T11:22:28UTC utils.py   INFO    : ['10.76.10.221']
2018-04-07T11:22:28UTC utils.py   INFO    : We have the following nodes to add to the inventory:
2018-04-07T11:22:28UTC utils.py   INFO    : ['10.76.10.218']
2018-04-07T11:22:28UTC utils.py   INFO    : Complete!
2018-04-07T11:22:28UTC scaler.py  INFO    : We've detected that we need to run ansible playbooks to scale up the cluster!
2018-04-07T11:22:28UTC scaler.py  INFO    : We will run the following ansible command:
2018-04-07T11:22:28UTC scaler.py  INFO    : ansible-playbook /usr/share/ansible/openshift-ansible/scaleup_wrapper.yml --extra-vars="{'scaling_category': 'master', 'target': 'provision_in_progress'}"
2018-04-07T11:22:28UTC scaler.py  INFO    : We have 1 ansible playbooks running!
2018-04-07T11:29:27UTC scaler.py  INFO    : - A process completed. We're parsing it...
2018-04-07T11:29:27UTC utils.py   INFO    : - [masters] playbook run results: {'failed': [], 'succeeded': [u'10.76.10.218'], 'unreachable': []}
2018-04-07T11:29:27UTC utils.py   INFO    : The json output logfile has been moved to /var/log/aws-quickstart-openshift-scaling.2018-4-7T1129
2018-04-07T11:29:27UTC scaler.py  INFO    : - complete! We have 0 to go...
2018-04-07T11:29:27UTC scaler.py  INFO    : Category: masters, Results: 1 / 0 / 0, (Succeeded / Unreachable / Failed)
2018-04-07T11:29:27UTC utils.py   INFO    : Nodes: [u'10.76.10.218'] have been permanately added to the Inventory under the masters category
2018-04-07T11:29:27UTC utils.py   INFO    : They've additionally been removed from the provision_in_progress category
2018-04-07T11:29:27UTC scaler.py  INFO    : ////////// End Script Invocation //////////

```

Inventory Status in Provision 
```
[OSEv3:children]
masters
nodes
etcd
new_nodes
new_masters
new_etcd

[OSEv3:vars]
ansible_ssh_user=root
openshift_deployment_type=openshift-enterprise
containerized=false
openshift_disable_check=package_version
openshift_master_cluster_method=native
openshift_router_selector='region=infra'
openshift_registry_selector='region=infra'
openshift_master_default_subdomain=**************
openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]
openshift_check_min_host_memory_gb=6
openshift_master_cluster_hostname=*************
openshift_master_cluster_public_hostname=********

[etcd]
10.76.12.45     openshift_hostname=10.76.12.45                                      
10.76.11.112    openshift_hostname=10.76.11.112                                      
10.76.10.240    openshift_hostname=10.76.10.240                                  

[masters]
10.76.11.183    openshift_hostname=10.76.11.183                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 
10.76.12.116    openshift_hostname=10.76.12.116                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 

[nodes]
10.76.11.183    openshift_hostname=10.76.11.183                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 
**10.76.10.221    openshift_hostname=10.76.10.221                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 
10.76.10.59     openshift_hostname=10.76.10.59                                      openshift_node_labels="{'application_node': 'yes', 'registry_node': 'yes', 'router_node': 'yes', 'region': 'infra', 'zone': 'default'} "  
10.76.12.209    openshift_hostname=10.76.12.209                                      openshift_node_labels="{'application_node': 'yes', 'registry_node': 'yes', 'router_node': 'yes', 'region': 'infra', 'zone': 'default'} "  
10.76.11.176    openshift_hostname=10.76.11.176                                      openshift_node_labels="{'application_node': 'yes', 'registry_node': 'yes', 'router_node': 'yes', 'region': 'infra', 'zone': 'default'} "  

[new_nodes]
10.76.10.218    openshift_hostname=10.76.10.218                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 

[provision_in_progress]
10.76.10.218

[new_masters]
10.76.10.218    openshift_hostname=10.76.10.218                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 

[new_etcd]

```

Inventory Final status:
```
[OSEv3:children]
masters
nodes
etcd
new_nodes
new_masters
new_etcd

[OSEv3:vars]
ansible_ssh_user=root
openshift_deployment_type=openshift-enterprise
containerized=false
openshift_disable_check=package_version
openshift_master_cluster_method=native
openshift_router_selector='region=infra'
openshift_registry_selector='region=infra'
openshift_master_default_subdomain=*************
openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]
openshift_check_min_host_memory_gb=6
openshift_master_cluster_hostname=************
openshift_master_cluster_public_hostname=*********

[etcd]
10.76.12.45     openshift_hostname=10.76.12.45                                    
10.76.11.112    openshift_hostname=10.76.11.112                                    
10.76.10.240    openshift_hostname=10.76.10.240                                

[masters]
10.76.11.183    openshift_hostname=10.76.11.183                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 
10.76.12.116    openshift_hostname=10.76.12.116                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 
10.76.10.218    openshift_hostname=10.76.10.218                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 

[nodes]
10.76.11.183    openshift_hostname=10.76.11.183                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 
10.76.10.221    openshift_hostname=10.76.10.221                                      openshift_node_labels="{'region': 'primary', 'zone': 'default'} " openshift_schedulable=false 
10.76.10.59     openshift_hostname=10.76.10.59                                      openshift_node_labels="{'application_node': 'yes', 'registry_node': 'yes', 'router_node': 'yes', 'region': 'infra', 'zone': 'default'} "  
10.76.12.209    openshift_hostname=10.76.12.209                                      openshift_node_labels="{'application_node': 'yes', 'registry_node': 'yes', 'router_node': 'yes', 'region': 'infra', 'zone': 'default'} "  
10.76.11.176    openshift_hostname=10.76.11.176                                      openshift_node_labels="{'application_node': 'yes', 'registry_node': 'yes', 'router_node': 'yes', 'region': 'infra', 'zone': 'default'} " 

[new_nodes]

[provision_in_progress]

[new_masters]

[new_etcd]

```
1. The new master is only in master section and not in nodes section
2. The deleted master is not removed from the initial node section 
3. The new Master fails to scale up due to the Unreachable SSH errors.

False Positive:
The python marks this scale up process as successful but the the new master host is still not functional
 
```
failed: [10.76.11.183] (item=10.76.10.221) => {"item": "10.76.10.221", "msg": "SSH Error: data could not be sent to remote host \"10.76.10.221\". Make sure this host can be reached over ssh", "unreachable": true}
```

```
PLAY RECAP ***********************************************************************************************************************************************************************************************
10.76.10.218               : ok=8    changed=0    unreachable=0    failed=0   
10.76.10.221               : ok=0    changed=0    unreachable=1    failed=0   
10.76.10.240               : ok=7    changed=0    unreachable=0    failed=0   
10.76.10.59                : ok=6    changed=0    unreachable=0    failed=0   
10.76.11.112               : ok=7    changed=0    unreachable=0    failed=0   
10.76.11.176               : ok=6    changed=0    unreachable=0    failed=0   
10.76.11.183               : ok=6    changed=0    unreachable=1    failed=0   
10.76.12.116               : ok=6    changed=0    unreachable=0    failed=0   
10.76.12.209               : ok=6    changed=0    unreachable=0    failed=0   
10.76.12.45                : ok=7    changed=0    unreachable=0    failed=0   
localhost                  : ok=11   changed=0    unreachable=0    failed=0 
```

When I remove the unreachable/deleted nodes from the inventory then when  I rerun the play books and fix the new master.

Please feel free to correct me if I am missing something .

**Comment by agamzn** [comment]
Adding commentary. 

The conversion from INI to YAML certainly helped this issue, but didn't fully resolve it. I ran into this today and I'm working on a fix now.

**Comment by philipakash** [comment]
Okay thanks for the update @agamzn .

**Comment by agamzn** [comment]
This was addressed in commit a5d77e6, and is now considered resolved.
